### PR TITLE
add attestation in the release job

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
- add attestation in the release job
/assign @puerco @saschagrunert @Verolop 
cc @kubernetes-sigs/release-engineering 

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
add attestation in the release job
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -35,14 +33,43 @@ jobs:
...
+      - name: Install tejolote
+        uses: puerco/release-actions/setup-tejolote@6c88cda6495b4415966e61f20798fb96a9081397 # main
+
+      - run: |
+          tejolote attest --artifacts github://kubernetes-sigs/bom/${{ steps.tag.outputs.tag_name }} github://kubernetes-sigs/bom/"${GITHUB_RUN_ID}" --output bom.intoto.json --sign
+
+      - name: Release
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # v0.1.15
+        with:
+          files: bom.intoto.json
```
Comment (puerco) [inline .github/workflows/release.yml:71]: I've been thinking that maybe we should start standardizing on a common name like `provenance.intoto.json` or something like that to help automation find it.

Let's think about some options next week :)

**Review by puerco — APPROVED** [review]
Awesome Carlos thank you 🤗

File: .github/workflows/release.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/release.yml:71]: ok sgtm

Comment (developer-guy) [reply .github/workflows/release.yml:71]: Another approach would be is converting it to a GitHub Action and provide an output to specify provenance name as SLSA Generators did ☝️

**Pull request merged**